### DISCUSSION TITLE
pyright registers didChangeWorkspaceFolders

### DISCUSSION THREAD

**Discussion by jdtsmith** [joaotavora/eglot#927] [General]
Pyright sends [didChangeWorkspaceFolders](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeWorkspaceFolders) registration, which eglot does not support, although it seems to be enabled:

> The notification is sent by default if both client capability `workspace.workspaceFolders` and the server capability `workspace.workspaceFolders.supported` are true; or if the server has registered itself to receive this notification.  

An example:

```
[server-request] (id:0) Wed Apr 20 20:40:08 2022:
(:jsonrpc "2.0" :id 0 :method "client/registerCapability" :params
	  (:registrations
	   [(:id "300f7ff6-a5e2-4269-97bb-60964362046c" :method "workspace/didChangeWorkspaceFolders" :registerOptions nil)]))
[client-reply] (id:0) Wed Apr 20 20:40:08 2022:
(:jsonrpc "2.0" :id 0 :result nil)
```

**Comment by nemethf** [comment]
Usually there's an explicit dynamicRegistration option in the client capabilities list for a given feature.  But not in this case.  So maybe the easiest fix would be to just ignore this server-request.
